src/Tribe/Linked_Posts.php

```diff
@@ -1120,21 +1124,48 @@ public function saved_linked_post_dropdown( $post_type, $current = null ) {
 		// Get the label to use in placeholder attrs.
 		$label = $this->get_create_or_find_labels( $post_type, $creation_enabled );
 
+		$render_option = function ( $option ) {
+			?>
+			<option <?php selected( $option['selected'] ) ?> value="<?php echo esc_attr( $option['id'] ); ?>">
+				<?php echo esc_html( $option['text'] ); ?>
+			</option>
+			<?php
+		};
+
 		if ( $linked_posts || $my_linked_posts ) {
-			echo '<input
-				type="hidden"
-				class="tribe-dropdown linked-post-dropdown"
-				name="' . esc_attr( $name ) . '"
-				id="saved_' . esc_attr( $post_type ) . '"
-				data-placeholder="' . $label . '"
-				data-search-placeholder="' . $label . '" ' .
-				( $creation_enabled ?
-				'data-freeform
+
+			?>
+			<select
+				class="tribe-dropdown linked-post-dropdown hide-before-select2-init"
+				name="<?php echo esc_attr( $name ); ?>"
+				id="saved_<?php echo esc_attr( $post_type ); ?>"
+				data-placeholder="<?php echo esc_attr( $label ); ?>"
+				data-search-placeholder="<?php echo esc_attr( $label ); ?>"
+				<?php if ( $creation_enabled ) : ?>
+				data-freeform
 				data-sticky-search
-				data-create-choice-template="' . __( 'Create: <b><%= term %></b>', 'the-events-calendar' ) . '" data-allow-html ' : '' ) .
-				'data-options="' . esc_attr( json_encode( $data ) ) . '"' .
-				( empty( $current ) ? '' : ' value="' . esc_attr( $current ) . '"' ) .
-			'>';
+				data-create-choice-template="<?php echo __( 'Create: <b><%= term %></b>', 'the-events-calendar' ); ?>"
+				data-allow-html
+				<?php endif; ?>
+			'>
+				<option value="-1" <?php selected( empty( $current ) ); ?>>
+					<?php echo esc_html( $label ); ?>
+				</option>
+				<?php if ( ! empty( $data[0]['children'] ) ) : ?>
+					<?php foreach ( $data as $group ) : ?>
+						<optgroup label="<?php echo esc_attr( $group['text'] ); ?>">
+							<?php foreach ( $group['children'] as $value ) : ?>
+								<?php $render_option( $value ); ?>
+							<?php endforeach; ?>
+						</optgroup>
+					<?php endforeach; ?>
+				<?php else: ?>
```

Comment:
* :no_entry_sign: Expected 1 space(s) after ELSE keyword; 0 found
* :no_entry_sign: Space between opening control structure and T_COLON is required

<sub>hash: 40e625c8f56618dcc7d0e4d0509a275e</sub>